realm/realm-library/src/main/java/io/realm/RealmObject.java

```diff
@@ -692,16 +691,16 @@ public final void removeAllChangeListeners() {
      * Gets {@link RealmConfiguration} from managed {@link RealmModel} instance such as {@link RealmObject} or {@link DynamicRealmObject}.
      *
      * @param model managed Realm object.
-     * @return {@link RealmConfiguration} of the Realm where the {@code model} belongs to.
-     * @throws IllegalArgumentException if the {@code model} is {@code null} or unmanaged.
+     * @return {@link RealmConfiguration} of the Realm where the {@code model} belongs to or {@code null} if the {@code model} is unmanaged.
+     * @throws IllegalArgumentException if the {@code model} is {@code null}.
      * @throws IllegalStateException if the model is deleted, the corresponding Realm of the model is closed or in an incorrect thread.
      */
     public static RealmConfiguration getConfiguration(RealmModel model) {
         if (model == null) {
             throw new IllegalArgumentException(MSG_NULL_OBJECT);
         }
         if (!(model instanceof RealmObjectProxy)) {
-            throw new IllegalArgumentException(MSG_NOT_MANAGED_OBJECT);
+            return null;
```

Comment:
No, but I'm not sure the other one is a good idea either. This at lest is more consistent with the behavior of our other methods: `isValid/isLoaded` that return values for unmanaged objects instead of crashing.